BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,413 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
+    Collection<LoadBalancer> allLoadBalancers = getAllLoadBalancers(baragonGroups);
+    try {
+      for (BaragonGroup baragonGroup : baragonGroups) {
+        if (baragonGroup.getSources().isEmpty()) {
+          LOG.debug("No traffic sources present for group {}", baragonGroup.getName());
+        } else {
+          Collection<LoadBalancer> elbsForBaragonGroup = getLoadBalancersByBaragonGroup(allLoadBalancers, baragonGroup);
+          Collection<BaragonAgentMetadata> baragonAgents = getAgentsByBaragonGroup(baragonGroup);
+
+
+          LOG.debug("Looking for TargetGroup for baragon group {}", baragonGroup.getName());
+          Optional<TargetGroup> maybeTargetGroup = guaranteeTargetGroupFor(baragonGroup);
+          if (maybeTargetGroup.isPresent()) {
+            TargetGroup targetGroup = maybeTargetGroup.get();
+            Collection<TargetDescription> targets = targetsInTargetGroup(targetGroup);
+
+            LOG.debug("Registering new instances for group {}", baragonGroup.getName());
+            guaranteeRegistered(targetGroup, targets, baragonAgents, elbsForBaragonGroup);
+
+            if (configuration.isPresent() && configuration.get().isDeregisterEnabled()) {
+              LOG.debug("De-registering old instances for group {}", baragonGroup.getName());
+              deregisterIfOld(baragonGroup, targetGroup, baragonAgents, targets);
+            }
+          } else {
+            LOG.debug("Not TargetGroup for Baragon Group {}", baragonGroup);
+          }
+
+          LOG.debug("ELB sync complete for group {}", baragonGroup);
+        }
+      }
+    } catch (AmazonClientException acexn) {
+      LOG.error("Could not retrieve elb information due to ELB client error {}", acexn);
+      exceptionNotifier.notify(acexn, ImmutableMap.of("groups", baragonGroups == null ? "" : baragonGroups.toString()));
+    } catch (Exception exn) {
+      LOG.error("Could not process ELB sync due to error {}", exn);
+      exceptionNotifier.notify(exn, ImmutableMap.of("groups", baragonGroups == null ? "" : baragonGroups.toString()));
+    }
+  }
+
+  private Collection<LoadBalancer> getAllLoadBalancers(Collection<BaragonGroup> baragonGroups) {
+    Set<String> trafficSources = new HashSet<>();
+    for (BaragonGroup baragonGroup : baragonGroups) {
+      for (TrafficSource trafficSource : baragonGroup.getSources()) {
+        if (trafficSource.getType() == TrafficSourceType.APPLICATION) {
+          trafficSources.add(trafficSource.getName());
+        }
+      }
+    }
+
+    Collection<LoadBalancer> loadBalancers = new HashSet<>();
+    DescribeLoadBalancersRequest loadBalancersRequest = new DescribeLoadBalancersRequest()
+        .withNames(trafficSources);
+    DescribeLoadBalancersResult result = elbClient.describeLoadBalancers(loadBalancersRequest);
+    String nextPage = result.getNextMarker();
+    loadBalancers.addAll(result.getLoadBalancers());
+
+    while (! nextPage.isEmpty()) {
```

Comment:
Just checking, will the aws API ever return a null for this, or is it just an empty string when there are no more pages? If it's possible there might be a null, I'd opt for `!Strings.isNullOrEmpty()` here